src/olympia/addons/utils.py

```diff
@@ -90,22 +91,27 @@ def get_addon_recommendations(guid_param, taar_enable):
     else:
         outcome = TAAR_LITE_OUTCOME_CURATED
     if not guids:
-        guids = TAAR_LITE_FALLBACKS
+        guids = get_filtered_fallbacks(guid_param)
     return guids, outcome, fail_reason
 
 
 def is_outcome_recommended(outcome):
     return outcome == TAAR_LITE_OUTCOME_REAL_SUCCESS
 
 
-def get_addon_recommendations_invalid():
+def get_addon_recommendations_invalid(current_guid=None):
     return (
-        TAAR_LITE_FALLBACKS,
+        get_filtered_fallbacks(current_guid),
         TAAR_LITE_OUTCOME_REAL_FAIL,
         TAAR_LITE_FALLBACK_REASON_INVALID,
     )
 
 
+def get_filtered_fallbacks(current_guid=None):
```

Comment:
It's unclear why we need to return only 4 items.  If this is a constraint on the FE we should handle it there. This just seems arbitrary to loop an entire list of static items and then slice the first 4 items.